src/wallet/use-cases/register-wallet/register-wallet.spec.ts

```diff
@@ -0,0 +1,120 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { BadRequestException, NotFoundException } from '@nestjs/common';
+
+import { RegisterWalletUseCase } from '@/wallet/use-cases';
+import { IWalletRepository } from '@/wallet/interfaces';
+import { SaveWalletDTO } from '@/wallet/dto';
+import { AccountType } from '@/shared/constants/enums';
+import { ListUserByIdUseCase } from '@/user/use-cases';
+import { FindBankByIdUseCase } from '@/bank/use-cases';
+import { Wallet } from '@/wallet/infra/entities';
+
+describe('RegisterWalletUseCase', () => {
+  let registerWalletUseCase: RegisterWalletUseCase;
+  let walletRepository: IWalletRepository;
+  let listUserByIdUseCase: ListUserByIdUseCase;
+  let findBankByIdUseCase: FindBankByIdUseCase;
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [
+        RegisterWalletUseCase,
+        {
+          provide: IWalletRepository,
+          useValue: {
+            create: jest.fn(),
+          },
+        },
+        {
+          provide: ListUserByIdUseCase,
+          useValue: {
+            execute: jest.fn(),
+          },
+        },
+        {
+          provide: FindBankByIdUseCase,
+          useValue: {
+            execute: jest.fn(),
+          },
+        },
+      ],
+    }).compile();
+
+    registerWalletUseCase = module.get<RegisterWalletUseCase>(
+      RegisterWalletUseCase,
+    );
+    walletRepository = module.get<IWalletRepository>(IWalletRepository);
+    listUserByIdUseCase = module.get<ListUserByIdUseCase>(ListUserByIdUseCase);
+    findBankByIdUseCase = module.get<FindBankByIdUseCase>(FindBankByIdUseCase);
+  });
+
+  it('should be able to register a new wallet', async () => {
+    listUserByIdUseCase.execute = jest.fn().mockResolvedValueOnce({});
+    findBankByIdUseCase.execute = jest.fn().mockResolvedValueOnce({});
+    walletRepository.create = jest.fn().mockResolvedValue({} as Wallet);
+
+    const saveWalletDTO: SaveWalletDTO = {
+      user_id: 'user_id',
+      bank_id: 'bank_id',
+      account_type: AccountType.CHECKING_ACCOUNT,
+      description: 'Descrição da carteira',
+    };
+
+    const result = await registerWalletUseCase.createWallet(saveWalletDTO);
+
+    expect(result).toBeDefined();
+    expect(walletRepository.create).toHaveBeenCalledWith(saveWalletDTO);
+  });
+
+  it('should not be able to return a wallet when bank id do not exist', async () => {
```

Comment:
O correto seria usar 'when bank id **DOES** not exist', pois o sujeito da sentença é singular e o verbo "to do" é utilizado como auxiliar.
- O "Do" é utilizado com "I", "you", "we", e "they" e com nomes no plural.
- O "Does" é utilizado com "he", "she", e "it" e com nomes no singular.

Neste cenário, você está referenciando o id do banco.

Arrumar a descrição do teste abaixo, também.

Reply:
Falta de atenção minha mesmo! Obrigado pela explicação Kayke, alterei aqui